manager/controllers/app/fybrikmodule_controller.go

```diff
@@ -54,22 +55,27 @@ func (r *FybrikModuleReconciler) Reconcile(ctx context.Context, req ctrl.Request
 
 	observedStatus := moduleContext.Status.DeepCopy()
 	moduleVersion := moduleContext.GetGeneration()
+	if len(moduleContext.Status.Conditions) == 0 {
+		moduleContext.Status.Conditions = []api.Condition{{Type: api.ReadyCondition, Status: corev1.ConditionUnknown, ObservedGeneration: 0}}
+	}
 
 	// check if module has been validated before or if validated module is outdated
-	if string(moduleContext.Status.ValidModule) == "" || observedStatus.ValidatedGeneration != moduleVersion {
+	condition := moduleContext.Status.Conditions[0]
```

Comment:
where is the condition type is checked? it seems to be updated later without checking the type.

Reply:
Condition type is not checked. Conditions are assumed to be static, i.e. having the same indices in the array over the run.

Reply:
OK. does it make sense to use `ReadyConditionIndex`  instead of 0?